jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py

```diff
@@ -357,6 +357,15 @@ def calculate_moment(self, add_data=True):
                 self.moment = moment
 
         # Reattach the WCS so we can load the result
+        if hasattr(w, 'celestial'):
+            # This is the FITS WCS case
+            data_wcs = getattr(w, 'celestial', None)
+        elif hasattr(w, 'to_fits_sip'):
+            # If it's a GWCS we pull out the celestial part
+            data_wcs = WCS(w.to_fits_sip())
```

Comment:
do we always want to do this?  For loading in actual data, we only convert to sip if requested by the user.  What change in specutils requires this?

Reply:
This is for the case where the coords are a GWCS, because GWCS doesn't have separable `celestial` and `spectral` attributes that you can just pull out individually (unless I've missed something...). This is the only way I found to get out just the 2D spatial solution from the 3D GWCS.

Reply:
Basically you _can_ make a SIPS approximation of the 3 dimensional cube, but you don't have to because you can just use the 3D GWCS. But this is easiest (only?) way to get the purely 2D spatial solution out of the GWCS that I know of.